### PR TITLE
Any rule

### PR DESCRIPTION
**Pull request by Sammers21**
_Closes #170

### CODE REVIEW

**Comment by Sammers21** [comment]
@g4s8, pay attention to the Pull Request.

**Comment by 0crat** [comment]
This pull request [#171](https://github.com/artipie/http/issues/171) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8708d1d3-1bb9-4527-9164-46ae180308ba); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat assign me

File: src/main/java/com/artipie/http/rt/RtRule.java [hunk 1]
```diff
@@ -55,7 +55,7 @@
     boolean apply(String line, Iterable<Map.Entry<String, String>> headers);
 
     /**
-     * Route by multiple rules.
+     * This rule is matched only when all of the rules is matched.
```
Comment (g4s8) [inline src/main/java/com/artipie/http/rt/RtRule.java:58]: @Sammers21 I'd deprecate this implementation and create new `All` to use same syntax for rules (`All`, `Any`)

**Comment by 0crat** [comment]
@0crat[/z](https://www.0crat.com/u/0crat) I see that you unassigned this issue; the order is still assigned to @g4s8[/z](https://www.0crat.com/u/g4s8) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/artipie/http/issues/171#issuecomment-644051245))

@g4s8 This pull request [#171](https://github.com/artipie/http/issues/171) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/0d650dac-869e-4f93-8cfb-9fc5651ecfee); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/http/rt/RtRule.java [hunk 2]
```diff
@@ -55,10 +55,20 @@
...
     /**
-     * Route by multiple rules.
+     * This rule is matched only when all of the rules are matched.
+     * This class is kept for backward compatibility reasons.
      * @since 0.5
+     * @deprecated use {@link All} instead
      */
-    final class Multiple implements RtRule {
+    @Deprecated
+    final class Multiple extends All {
```
Comment (g4s8) [inline src/main/java/com/artipie/http/rt/RtRule.java:64]: @Sammers21 if you don't put a constructor for this class, Java will generate default empty constructor which calls `All` varargs constructor with empty array:
```
final class Multiple extends All {
  Multiple() {
    super(new RtRule[0]);
  }
}
```
So existing code won't be able to access other constructors from `All`.

**Comment by 0crat** [comment]
Job was finished in 17 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**